scripts/tests/test_validate_docstrings.py

```diff
@@ -594,7 +603,82 @@ def test_bad_generic_functions(self, func):
                      marks=pytest.mark.xfail)
     ])
     def test_bad_examples(self, capsys, klass, func, msgs):
-        validate_one(self._import_path(klass=klass, func=func))  # noqa:F821
-        err = capsys.readouterr().err
+        result = validate_one(self._import_path(klass=klass, func=func))  # noqa:F821
         for msg in msgs:
-            assert msg in err
+            assert msg in ' '.join(result['errors'])
```

Comment:
Minor nit but what does this add? Can't we just do the normal inclusion check against the list instead of joining into a string?

Reply:
`msg in result['errors']` is different than this, for example `'a' in 'foo bar'` is `True`, but `a in ['foo', 'bar']` is `False`.

I think the `.join()` is simpler than another loop, is it what you would do, or you were thinking on the previous case?

Reply:
No but I thought `msg` would explicitly match one of the `errors`; will take a look more deeply on next review if that assumption is incorrect